internal/clientconn/conninfo/conn_info.go

```diff
@@ -53,6 +65,33 @@ func (connInfo *ConnInfo) SetAuth(username, password string) {
 	connInfo.password = password
 }
 
+// Cursor returns the cursor value stored.
+// We use "db.collection" as the key to get the cursor.
+func (connInfo *ConnInfo) Cursor(key string) iterator.Interface[int, any] {
+	connInfo.curRW.RLock()
+	defer connInfo.curRW.RUnlock()
+
+	return connInfo.cursor[key]
+}
+
+// SetCursor stores the cursor value.
+// We use "db.collection" as the key to store the cursor.
+func (connInfo *ConnInfo) SetCursor(key string, cursor iterator.Interface[int, any]) {
+	connInfo.curRW.Lock()
+	defer connInfo.curRW.Unlock()
+
+	connInfo.cursor[key] = cursor
+}
+
+// RemoveCursor removes the cursor value stored.
+// We use "db.collection" as the key to remove the cursor.
+func (connInfo *ConnInfo) RemoveCursor(key string) {
+	connInfo.curRW.Lock()
+	defer connInfo.curRW.Unlock()
+
+	delete(connInfo.cursor, key)
```

Comment:
What happens if we delete an iterator that wasn't closed? Is there a potential memory leak somewhere?

Reply:
Added `cursor.Close`. I don't think we have to test it now as we iterate over `types.Array` and `Close` method doing nothing.